File: backend/test_code/test_code1.ts
export {}; // This prevents the 'Cannot redeclare block-scoped variable' error!

// The URL where your FastAPI server is running
const API_URL = "http://127.0.0.1:8000/api/heal-code";

// 1. Define the exact structure we expect to send
interface RepairRequest {
    language: string;
    error_message: string;
    source_code: string;
}

// 2. We simulate a broken Cypress E2E test
const brokenCypressTest: RepairRequest = {
    language: "typescript",
    error_message: "AssertionError: Timed out retrying after 4000ms: Expected to find element: `[data-cy=login-btn]`, but never found it.",
    source_code: `
describe('Authentication Flow', () => {
    it('should log the user in successfully', () => {
        cy.visit('/login');
        
        // The frontend team changed the data attribute to data-testid="login-button"
        // This causes the E2E test to timeout and fail.
        cy.get('[data-cy=login-btn]').click();
        
        cy.get('input[name=username]').type('testuser');
        cy.get('input[name=password]').type('password123');
        cy.get('form').submit();
        
        cy.url().should('include', '/dashboard');
    });
});
`
};

// 3. The async function to call your Python backend
async function runCypressTest() {
    console.log("🚀 Sending broken Cypress test to Guardian Auto-Healer...");
    console.log("-".repeat(60));
    
    try {
        const response = await fetch(API_URL, {
            method: "POST",
            headers: {
                "Content-Type": "application/json",
            },
            body: JSON.stringify(brokenCypressTest),
        });

        if (response.ok) {
            const aiResult = await response.json();
            
            console.log("✅ HEALING SUCCESSFUL!\n");
            console.log("💡 AI Explanation:");
            console.log(`   ${aiResult.explanation}`);
            console.log(`   Confidence Score: ${aiResult.confidence}\n`);
            
            console.log("🛠️ FIXED CODE:");
            console.log(aiResult.fixed_code);
        } else {
            console.log(`❌ Server Error: ${response.status}`);
            const errorText = await response.text();
            console.log(errorText);
        }
    } catch (error) {
        console.log("❌ Error: Could not connect to the server.");
        console.log("Make sure your FastAPI server is running with: uvicorn main:app --reload");
        console.error(error);
    }
}

// Execute the test
runCypressTest();

File: backend/test_code/test_code1.py
import requests
import json

# The URL where your FastAPI server is running
API_URL = "http://127.0.0.1:8000/api/heal-code"

# 1. We simulate a broken test scenario (e.g., UI changed, ID is missing)
broken_test_data = {
    "language": "python",
    "error_message": "NoSuchElementException: Message: no such element: Unable to locate element: {\"method\":\"css selector\",\"selector\":\"#old-login-btn\"}",
    "source_code": """
def test_user_login(driver):
    # Navigating to the login page
    driver.get("https://example.com/login")
    
    # Trying to click the login button using the old, broken ID
    login_button = driver.find_element(By.ID, "old-login-btn")
    login_button.click()
    
    assert "Dashboard" in driver.title
"""
}

def run_test():
    print("🚀 Sending broken code to Guardian Auto-Healer...")
    print("-" * 50)
    
    try:
        # 2. Send the POST request to your FastAPI backend
        response = requests.post(API_URL, json=broken_test_data)
        
        # 3. Check if the server responded successfully
        if response.status_code == 200:
            ai_result = response.json()
            
            print("✅ HEALING SUCCESSFUL!\n")
            print("💡 AI Explanation:")
            print(f"   {ai_result.get('explanation')}")
            print(f"   Confidence Score: {ai_result.get('confidence')}\n")
            
            print("🛠️ FIXED CODE:")
            print(ai_result.get('fixed_code'))
            
        else:
            print(f"❌ Server Error: {response.status_code}")
            print(response.text)
            
    except requests.exceptions.ConnectionError:
        print("❌ Error: Could not connect to the server.")
        print("Make sure your FastAPI server is running with: uvicorn main:app --reload")

if __name__ == "__main__":
    run_test()